core/utils.js

```diff
@@ -587,6 +587,47 @@ Blockly.utils.getBlockTypeCounts = function(block, opt_stripFollowing) {
   return typeCountsMap;
 };
 
+var resizeFromKeyboard = true;
+var initialWidth = 0;
+var debounceTimeout = null;
+
+/**
+ * Called in the resize event handler to sense if the source
+ * is from an on screen keyboard or user.
+ * @param {!Blockly.WorkspaceSvg} workspace Any workspace in the SVG.
+ */
+Blockly.utils.resizeTracker = function(workspace) {
+  if (debounceTimeout) {
+    clearTimeout(debounceTimeout);
+  }
+  debounceTimeout = setTimeout(function(){
+    resizeFromKeyboard = true;
+    initialWidth = workspace.width;
+  }, 1000);
+
+  if (resizeFromKeyboard) {
+    // If the on-screen keyboard is causing the resize, only the
+    // height will change.
+    if (workspace.width != initialWidth) {
+      resizeFromKeyboard = false;
+    }
+  }
+}
+
+/**
+ * Use during resize to see if the source of the resize
+ * is from an on-screen keyboard appearing.
+ */
+Blockly.utils.isOnScreenKeyboardResize = function() {
+  if (document.activeElement.className == "blocklyHtmlInput" && resizeFromKeyboard) {
```

Comment:
should probably use "hasClass" https://github.com/microsoft/pxt-blockly/blob/develop/core/utils/dom.js#L164 since the className is a string of all classes on the element & could change in the future

Reply:
fixed! thanks shannon!